src/MiNET/MiNET/Net/Packet.cs

```diff
@@ -1577,7 +1598,37 @@ public Skin ReadSkin()
 			skin.IsPersonaSkin = ReadBool();
 			skin.Cape.OnClassicSkin = ReadBool();
 			skin.Cape.Id = ReadString();
-			ReadString(); // some unique skin id
+			ReadString(); // fullSkinId
+			skin.ArmSize = ReadString();
+			skin.SkinColor = ReadString();
+			int personaPieces = ReadInt();
+			skin.PersonaPieceCount = personaPieces;
```

Comment:
this seems useless? `PersonaPieces.Count()` should be enough

Reply:
But I need to read the Integer at one point anyways, so why not using it? Fairly, I could simplify this by setting the skin property to the read value directly and iterating over that instead of saving it in an extra variable. Or do I understand something wrong?

Reply:
On second look there are a lot more issues with this PR than just this useless assignment.

My point here is that `PersonaPieceCount` field is useless because you can just use `PersonaPieces.Count` to get the number of pieces after the effect. This number is only useful during decoding.

Reply:
Ah I get what you mean, yes, will fix that soon